pallets/redeem/src/lib.rs

```diff
@@ -822,6 +826,11 @@ impl<T: Config> Pallet<T> {
 				&confiscated_collateral,
 			)?;
 
+			CancelledRedeemAmount::<T>::insert(
+				redeem_id,
+				(confiscated_collateral.amount(), confiscated_collateral.currency()),
+			);
+
```

Comment:
Somehow I fear that this is not accurate and we should not do this. 
The audit ticket was about issues with `transfer_funds_saturated()` and not about the `transfer_funds()` case. I am not 100% sure though, but I'd prefer if we remove this and add an additional check below.